apm-agent-plugins/apm-aws-sdk/apm-aws-sdk-1-plugin/src/main/java/co/elastic/apm/agent/awssdk/v1/helper/SQSHelper.java

```diff
@@ -0,0 +1,234 @@
+import com.amazonaws.AmazonWebServiceRequest;
+import com.amazonaws.Request;
+import com.amazonaws.http.ExecutionContext;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.MessageAttributeValue;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.ReceiveMessageResult;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequest;
+import com.amazonaws.services.sqs.model.SendMessageBatchRequestEntry;
+import com.amazonaws.services.sqs.model.SendMessageRequest;
+
+import javax.annotation.Nullable;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+public class SQSHelper extends AbstractSQSInstrumentationHelper<Request<?>, ExecutionContext, Message> implements TextHeaderSetter<Map<String, MessageAttributeValue>> {
+
+    @Nullable
+    private static final SQSHelper INSTANCE = new SQSHelper(GlobalTracer.requireTracerImpl());
+
+    public static SQSHelper getInstance() {
+        return INSTANCE;
+    }
+
+    protected SQSHelper(ElasticApmTracer tracer) {
+        super(tracer, SdkV1DataSource.getInstance());
+    }
+
+
+    public void propagateContext(Span span, AmazonWebServiceRequest request) {
+        if (request instanceof SendMessageRequest) {
+            SendMessageRequest sendMessageRequest = (SendMessageRequest) request;
+            span.propagateTraceContext(sendMessageRequest.getMessageAttributes(), this);
+        } else if (request instanceof SendMessageBatchRequest) {
+            SendMessageBatchRequest sendMessageBatchRequest = (SendMessageBatchRequest) request;
+            for (SendMessageBatchRequestEntry entry : sendMessageBatchRequest.getEntries()) {
+                span.propagateTraceContext(entry.getMessageAttributes(), this);
+            }
+        }
+    }
+
+    public void setMessageAttributeNames(ReceiveMessageRequest receiveMessageRequest) {
+        List<String> messageAttributeNames = receiveMessageRequest.getMessageAttributeNames();
+        if (!messageAttributeNames.contains(ATTRIBUTE_NAME_ALL) && !messageAttributeNames.contains(TraceContext.W3C_TRACE_PARENT_TEXTUAL_HEADER_NAME)) {
+            messageAttributeNames.add(TraceContext.W3C_TRACE_PARENT_TEXTUAL_HEADER_NAME);
+        }
+
+        if (!messageAttributeNames.contains(ATTRIBUTE_NAME_ALL) && !messageAttributeNames.contains(TraceContext.TRACESTATE_HEADER_NAME)) {
+            messageAttributeNames.add(TraceContext.TRACESTATE_HEADER_NAME);
+        }
+
+        List<String> attributeNames = receiveMessageRequest.getAttributeNames();
+        if (!attributeNames.contains(ATTRIBUTE_NAME_SENT_TIMESTAMP)) {
+            attributeNames.add(ATTRIBUTE_NAME_SENT_TIMESTAMP);
+        }
+    }
+
+    @Override
+    protected String getMessageBody(Message sqsMessage) {
+        return sqsMessage.getBody();
+    }
+
+    @Nullable
+    protected Long getMessageAge(Message message) {
```

Comment:
Is this Long rather than long for a reason? The null check could be replaced with Long.MIN_VALUE then we wouldn't have any autoboxing concern (escape analysis might get rid of the box, but it might not on some JVMs)

Reply:
replaced it with `long` now